src/scene/materials/standard-material-options-builder.js

```diff
@@ -348,7 +349,7 @@ class StandardMaterialOptionsBuilder {
                 if (stdMat[uname] === 1 && !hasUv1) allow = false;
                 if (allow) {
                     options[mname] = !!stdMat[mname];
-                    options[iname] = stdMat[mname].name;
+                    options[iname] = stdMat[mname].id;
```

Comment:
actually I'm not sure I like new version, or old version, as this would make a unique shader for each unique texture, causing a lot more shaders getting compiled if material simply uses a different texture.

Reply:
enable tracing of shader creations this way and test with / without this line, it will likely generate more shaders if I understand it correctly
`pc.Tracing.set(pc.TRACEID_SHADER_ALLOC, true);`

Reply:
I agree, I was thinking how this could be prevented, and I figured this would work:

```js
                    const mapId = stdMat[mname].id;
                    let identifier = textureMap[mapId];
                    if (identifier === undefined) {
                        textureMap[mapId] = textureMappingCounter.value;
                        identifier = textureMappingCounter.value;
                        textureMappingCounter.value++;
                    }
                    ...
                    options[iname] = identifier;

```

Where textureMap and textureMappingCounter is shared over all textures. This way, the mapping counter will be incremented deterministically for each material, so if there is an identical mapping between two changes, there is no change in the shader.

Reply:
I'm not sure I understand. It seems that you're creating a map, which maps unique texture.id into some numerical id that gets incremented for each texture this code encounters?

The problem with solving this on the options-builder side is that it might not match what the actual shader generator does I think. Some chunks (non-updated user chunks, or lightmapper chunk) are not sharing their samplers, and this won't handle it? This could cause incorrect shader being pulled from the cache, and hard to track incorrect textures bug introduced.

Reply:
Yeah, what this will do is to avoid accidentally generating a new shader if the texture sharing is identical but the texture ids mismatch. With this, the ids can't mismatch because we generate the identifiers in the function when going over all the supported textures in the material.

This should make sure the identifiers are deterministic if the texture changes but the sharing remains identical. 

For chunks that don't share their samplers, do we need to worry about it? As long as they define the samplers, don't use the $SAMPLER macro, and provide a texture it shouldn't be an issue?

Reply:
yep, ok, sounds good to me I think.